src/Signup_Command.php

```diff
@@ -0,0 +1,272 @@
+ *     | 1         | bobuser    | bobuser@example.com | 2024-03-13 05:46:53 | 1      | 7320b2f009266618 |
+ *     | 2         | johndoe    | johndoe@example.com | 2024-03-13 06:24:44 | 0      | 9068d859186cd0b5 |
+ *     +-----------+------------+---------------------+---------------------+--------+------------------+
+ *
+ *     # Activate signup.
+ *     $ wp user signup activate 2
+ *     Success: Signup 2 activated. Password: bZFSGsfzb9xs
+ *
+ *     # Delete signup.
+ *     $ wp user signup delete 3
+ *     Success: Signup 3 deleted.
+ *
+ * @package wp-cli
+ */
+class Signup_Command extends CommandWithDBObject {
+
+	protected $obj_type = 'signup';
+
+	protected $obj_id_key = 'signup_id';
+
+	protected $obj_fields = [
+		'signup_id',
+		'user_login',
+		'user_email',
+		'registered',
+		'active',
+		'activation_key',
+	];
+
+	private $fetcher;
+
+	public function __construct() {
+		$this->fetcher = new SignupFetcher();
+	}
+
+	/**
+	 * Lists signups.
+	 *
+	 * [--<field>=<value>]
+	 * : Filter the list by a specific field.
+	 *
+	 * [--field=<field>]
+	 * : Prints the value of a single field for each signup.
+	 *
+	 * [--fields=<fields>]
+	 * : Limit the output to specific object fields.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - ids
+	 *   - json
+	 *   - count
+	 *   - yaml
+	 * ---
+	 *
+	 * ## AVAILABLE FIELDS
+	 *
+	 * These fields will be displayed by default for each signup:
+	 *
+	 * * signup_id
+	 * * user_login
+	 * * user_email
+	 * * registered
+	 * * active
+	 * * activation_key
+	 *
+	 * These fields are optionally available:
+	 *
+	 * * domain
+	 * * path
+	 * * title
+	 * * activated
+	 * * meta
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     # List signup IDs.
+	 *     $ wp user signup list --field=signup_id
+	 *     1
+	 *
+	 *     # List all signups.
+	 *     $ wp user signup list
+	 *     +-----------+------------+---------------------+---------------------+--------+------------------+
+	 *     | signup_id | user_login | user_email          | registered          | active | activation_key   |
+	 *     +-----------+------------+---------------------+---------------------+--------+------------------+
+	 *     | 1         | bobuser    | bobuser@example.com | 2024-03-13 05:46:53 | 1      | 7320b2f009266618 |
+	 *     | 2         | johndoe    | johndoe@example.com | 2024-03-13 06:24:44 | 0      | 9068d859186cd0b5 |
+	 *     +-----------+------------+---------------------+---------------------+--------+------------------+
+	 *
+	 * @subcommand list
+	 *
+	 * @package wp-cli
+	 */
+	public function list_( $args, $assoc_args ) {
+		global $wpdb;
+
+		if ( isset( $assoc_args['fields'] ) ) {
+			$assoc_args['fields'] = explode( ',', $assoc_args['fields'] );
+		} else {
+			$assoc_args['fields'] = $this->obj_fields;
+		}
+
+		$signups = array();
+
+		$results = $wpdb->get_results( "SELECT * FROM $wpdb->signups", ARRAY_A );
```

Comment:
Should we add some limit here?

Reply:
We don't have a limit by default in `wp post list`, so I don't think we need a limit here.

We should allow the list to be limited, though

Reply:
@danielbachhuber like `--limit` option? Defaulting to 100 or something.

Reply:
@ernilambar Yes, a `--limit=` option or similar (maybe `--per_page=`). It can default to none, though.